bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -1411,4 +1418,50 @@ private void updateThingFromLegacy() {
             }
         }
     }
+
+    /**
+     * Update the thing's semantic equipment tag. The main determinant for the equipment type is the supported channels.
+     * For lights use the product archetype to split between light bulbs and strip lights. Rooms and Zones are
+     * considered to be (groups of) light bulbs.
+     */
+    private void updateEquipmentTag() {
+        if (!disposing) {
+            int sensorCount = 0;
+            SemanticTag equipmentTag = Equipment.LIGHTBULB; // default;
+
+            if (Set.of(ResourceType.ROOM, ResourceType.ZONE).contains(thisResource.getType())) {
+                equipmentTag = Equipment.ZONE;
+            }
+            if (thisResource.getProductData() instanceof ProductData productData
+                    && STRIPLIGHT_ARCHETYPES.contains(productData.getProductArchetype())) {
+                equipmentTag = Equipment.LIGHT_STRIP;
+            }
+            if (thing.getChannel(CHANNEL_2_BUTTON_LAST_EVENT) != null) {
+                equipmentTag = Equipment.BUTTON;
+            }
+            if (thing.getChannel(CHANNEL_2_ROTARY_STEPS) != null) {
+                equipmentTag = Equipment.DIAL;
+            }
+            if (thing.getChannel(CHANNEL_2_SECURITY_CONTACT) != null) {
+                equipmentTag = Equipment.CONTACT_SENSOR;
+            }
+            if (thing.getChannel(CHANNEL_2_MOTION) != null) {
+                sensorCount++;
+                equipmentTag = Equipment.MOTION_DETECTOR;
+            }
+            if (thing.getChannel(CHANNEL_2_LIGHT_LEVEL) != null) {
+                sensorCount++;
+                equipmentTag = Equipment.ILLUMINANCE_SENSOR;
+            }
+            if (thing.getChannel(CHANNEL_2_TEMPERATURE) != null) {
+                sensorCount++;
+                equipmentTag = Equipment.TEMPERATURE_SENSOR;
+            }
+            if (sensorCount > 1) {
+                equipmentTag = Equipment.SENSOR;
```

Comment:
I'm not sure I fully agree to this logic. Would you say a Hue motion sensor should be considered as a `Sensor` rather than a `MotionDetector`, because it also measures temperature? I agee that makes it a "multi sensor", but motion detection seems to be the primary use-case. I have to admit that I'm behind with all of this, so I don't know if anything was clearly established for this case.

Reply:
I am not sure if the primary function is motion detector. I admit the temperature sensing is a side show. However one (other) primary function is the illuminace sensor. Some people use the motion part of the sensor to turn on lights when motion is detected; but others use the illuminance part to turn on lights when it gets dark; and others use both..